### PR TITLE
building: Exchange TA HSM signing step with openssl command

### PR DESCRIPTION
**Pull request by jclsn**
The pkcs11-tool approach had issues and is exchanged with the openssl command for conformity.

### CODE REVIEW

**Comment by jbech-linaro** [comment]
`Acked-by: Joakim Bech <joakim.bech@linaro.org>`

File: building/trusted_applications.rst [hunk 1]
```diff
@@ -500,20 +500,23 @@ OP-TEE's pkcs11 in-tree TA is used.
...
-       -pkeyopt rsa_pss_saltlen:digest -pkeyopt rsa_mgf1_md:sha256 | \
-   base64 > $UUID.sig
+    base64 --decode $UUID.dig | \
+    openssl pkeyutl -sign -inkey $TA_SIGN_KEY \
+        -pkeyopt digest:sha256 -pkeyopt rsa_padding_mode:pss \
+        -pkeyopt rsa_pss_saltlen:digest -pkeyopt rsa_mgf1_md:sha256 | \
+    base64 > $UUID.sig
 
-or with ``pkcs11-tool`` using a Nitrokey HSM
+or using a Nitrokey HSM (assuming a working openssl configuration for the PKCS11 engine is present)
```
Comment (jbech-linaro) [inline building/trusted_applications.rst:509]: s/openssl/OpenSSL/

**Comment by jbech-linaro** [comment]
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

**Comment by jclsn** [comment]
@jbech-linaro Should I amend all commits accordingly?

**Comment by jbech-linaro** [comment]
@jclsn I think these could go into a single commit, so squash the patches here, add the review-by tags etc and force push to the same branch. After that I'll merge this.

**Comment by jclsn** [comment]
I did squash them some time ago. Merge it?

**Comment by jbech-linaro** [comment]
> I did squash them some time ago. Merge it?

Yes, I'll merge right away. Thanks for the reminder.

**Pull request merged**